### PR TITLE
IP-612 Provide ability to total net money on array of Price instances

### PR DESCRIPTION
**Pull request by bartszczepaniak**
# Overview

This `PR` implements ability to total net money on Array of `Money` instances. 

## [IP-612]( https://izettle.atlassian.net/browse/IP-612)

### CODE REVIEW

File: Worth/Money/Money.swift [hunk 1]
```diff
@@ -58,3 +58,34 @@ extension Money: Equatable {
         return isAmountEqual && isCurrencyEqual
     }
 }
+
+public extension Array where Element == Money {
+
+    // MARK: - Public Functions
+
+    /// Extracts and `Array` of `Money` instances associated with given `Currency`.
```
Comment (Flinted) [inline Worth/Money/Money.swift]: `and` -> `an`

Comment (bartszczepaniak) [reply Worth/Money/Money.swift]: 👍

File: WorthTests/Unit Tests/MoneyTests.swift [hunk 2]
```diff
@@ -50,4 +50,50 @@ internal final class MoneyTests: XCTestCase {
...
+            Money(amount: 1, currency: gbp),
+            Money(amount: 1, currency: gbp),
+            Money(amount: 1, currency: usd),
+            Money(amount: 2, currency: gbp),
+            Money(amount: -1, currency: gbp),
+            Money(amount: 1, currency: gbp),
+            Money(amount: -1, currency: usd),
+            Money(amount: 0, currency: gbp)
+        ]
+        var money: [Money] = []
```
[the remark this answers is not in the document]
Comment (bartszczepaniak) [reply WorthTests/Unit Tests/MoneyTests.swift]: You are absolutely correct, this is leftover code and we should rename `variations` to `money` or keep it `variations` but remove new array of `money`. 👍

File: WorthTests/Unit Tests/MoneyTests.swift [hunk 3]
```diff
@@ -50,4 +50,50 @@ internal final class MoneyTests: XCTestCase {
...
```
[the remark this answers is not in the document]
Comment (bartszczepaniak) [reply WorthTests/Unit Tests/MoneyTests.swift]: same 👍

File: Worth/Money/Money.swift [hunk 4]
```diff
@@ -58,3 +58,34 @@ extension Money: Equatable {
...
     }
 }
+
+public extension Array where Element == Money {
+
+    // MARK: - Public Functions
+
+    /// Extracts an `Array` of `Money` instances associated with given `Currency`.
+    ///
+    /// - Parameter currency: `Currency` instance used to filter asociated `Money` instances.
```
[the remark this answers is not in the document]
Comment (bartszczepaniak) [reply Worth/Money/Money.swift]: 👍

File: Worth/Money/Money.swift [hunk 5]
```diff
@@ -58,3 +58,34 @@ extension Money: Equatable {
...
+    public func allMoney(with currency: Currency) -> [Money] {
+        let money = self.filter {
+            $0.currency == currency
+        }
+        return money
+    }
+
+    /// Provides total amount of `Money` instances with given `Currency`.
+    ///
+    /// - Parameter currency: `Currency` instance used to filter asociated `Money` instances.
```
[the remark this answers is not in the document]
Comment (bartszczepaniak) [reply Worth/Money/Money.swift]: 👍

**Pull request merged**